src/Disks/DiskCache.cpp

```diff
@@ -652,12 +685,17 @@ class CacheableMultipartReadBufferDecorator : public ReadBuffer
     {
         ProfileEvents::increment(ProfileEvents::DiskCacheRequestsIn, 1);
         log = &Poco::Logger::get("CacheableMultipartReadBufferDecorator");
-        ensureFrom();
     }
 
     void set(BufferBase::Position ptr, size_t size, size_t offset) override
     {
-        from->BufferBase::set(ptr, size, offset);
+        use_external_buffer = true;
```

Comment:
Is the meaning of this variable is the same as in all other buffers, where it was added?
If yes, then why explicitly setting it to true? It should only be true, if async reads are turned on by according setting.
If not, I think it should be named differently because it will confuse readers.

Reply:
It's used in same case, but means "someone sets external buffer with `set()` method". I'll rename this variable.